### PR TITLE
Do not search for items installed with Rakudo

### PR DESCRIPTION
**Pull request by nbrown**
Currently, when I use zef to install distributions that depend on things distributed with rakudo (e.g., nqp), zef fails to install them because it cannot find the dependencies to install. This patch filters them out of the dependencies that zef attempts to install.

### CODE REVIEW

**Comment by tony-o** [comment]
@nbrown what distribution are you attempting to install?

**Comment by ugexe** [comment]
nqp should not be listed as a dependency because it is a pragma and not a distribution (so it should never be listed as a dependency). This can be confirmed because this problem does not manifest for actual distributions that do come installed with rakudo (such as `Test`)

The issue was resolved via the author of Terminal::ANSIColor removing `nqp` from the depends list, allowing Task::Star to install properly. However this also requires you to `zef update` to refresh your package list to reflect this change - otherwise you will continue to get this failure due to the stale package list still serving the 'nqp' dependency (see: https://github.com/ugexe/zef/issues/113).

I'm not sure this PR will actually filter the way you think - `@!ignore` contains strings but you're trying to match them against a spec object. In any case, as mentioned in the link above for issue 113, the onus for a proper dependency listing is now on the module author because there is no way for zef to get a list of available pragmas at runtime.

Basically if you get an install error for nqp you should send a PR to the module including nqp as a dependency to remove it. If they already remove it (which should be the case) then you simply need to `zef update`

**Comment by nbrown** [comment]
Fair enough. I was trying to exclude these in this fix because some of the items in Task::Star depend on nqp and this seemed to be how panda handles it.
